### PR TITLE
support login session, query payload, and multi events

### PR DESCRIPTION
**Pull request by ryli17**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,6 +165,95 @@ You can use `user`, `password` options to provide authentication information.
...
+```
+<source>
+	@type http_pull
+
+	tag status
+	url https://localhost:8080
+	path search
+	interval 1s
+
+	login_path lobin
```
Comment (shadowcolossus192) [inline README.md]: _lobin_?

File: README.md [hunk 2]
```diff
@@ -161,6 +165,95 @@ You can use `user`, `password` options to provide authentication information.
...
+	login_payload {"username":"tester","password":"drowssaP"}
+	verify_ssl false
```
Comment (shadowcolossus192) [inline README.md:185]: Do we want to use ssl for login validation?

File: README.md (hunk 1 above)
Comment (lakshman-battini) [inline README.md]: @ryli17  typo here at login_path.

File: README.md [hunk 3]
```diff
@@ -161,6 +165,95 @@ You can use `user`, `password` options to provide authentication information.
...
+</source>
+```
+
+### Encapsulated items
+
+If the expected items are encapsulated in json structure, for example,
+
+```
+{"meta": {"server": "localhost"},
+  "iterms": [
```
Comment (lakshman-battini) [inline README.md]: @ryli17 event_key mismatch between the example json and config.

File: README.md (hunk 1 above)
Comment (ryli17) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (ryli17) [reply README.md]: Done

**Review by shadowcolossus192 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (ryli17) [reply README.md:185]: This plugin is pretty generic. In particular deployment, we can enable or disable verify_ssl.

File: README.md (hunk 1 above)
Comment (ryli17) [reply README.md]: fixed.

**Pull request merged**